### DISCUSSION TITLE
Please add the getRemoteAddress to the HttpRequest object

### DISCUSSION THREAD

**Discussion by dioptre** [uNetworking/uWebSockets.js#215] [Q&A] [answered]
It's possible to get this from forwarded headers, but I think we should have this too.

The res.getRemoteAddress is not accurate.

**Comment by ghost** [comment]
You already have res.getRemoteAddress(), it returns the IP address in either 4 or 16 bytes ArrayBuffer

**Comment by dioptre** [comment]
This never returns the correct client address for me.

**Comment by sirganya** [comment]
If you use this
`const buffer = Buffer.from(message)`
You'll get an array of the digits making up the IP address. 
You can join them with 
`const ip = buffer.join('.');`
You may have to do some more manipulation to get it how you like it but I'd recommend reading about array buffers first as they're how data will be passed to you.

**Comment by ghost** [comment]
I'm pretty sure it is correct right now

**Comment by ghost** [comment, marked as answer]
res.getRemoteAddressAsText()
